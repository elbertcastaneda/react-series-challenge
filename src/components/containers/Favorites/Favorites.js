import React, { Component } from 'react';
import { FavoritesContainer } from './Favorites.styled';
import { GiphyList, InputSearch, ErrorWatcher } from '../../presentational';
import { connect } from 'react-redux';
import favoriteGifsActions from '../../../actions/favoriteGifs';
const { creators } = favoriteGifsActions;
const { getFavoriteGifsRequested } = creators;
class Favorites extends Component {
  componentDidMount () {
    const { getFavoriteGifsRequested } = this.props;
    getFavoriteGifsRequested();
  }

  searchGifs (input) {
    const { getFavoriteGifsRequested } = this.props;
    return setTimeout(() => {
      getFavoriteGifsRequested({query: input.value});
    }, 250);
  }

  render() {
    const {favoriteGifs, getFavoriteGifsRequested} = this.props;
    console.log('​Favorites -> render -> favoriteGifs', favoriteGifs);
    return (
      <FavoritesContainer>
        <InputSearch 
          defaultValue={favoriteGifs.query}
          placeholder='Search imagesin my favorites ...' 
          onChange={({target: input}) => {
              if (this.writtingTimmer !== null) {
                clearTimeout(this.writtingTimmer);
                this.writtingTimmer = null;
              }
              this.writtingTimmer = this.searchGifs(input);
          }} />
        <GiphyList 
          dataSource={favoriteGifs}
          onClickFirstPage={() => getFavoriteGifsRequested({offset:1})}
          onClickPreviousPage={() => getFavoriteGifsRequested({movePrevious:true})}
          onClickNextPage={() => getFavoriteGifsRequested({moveNext:true})}
          onClickLastPage={() => getFavoriteGifsRequested({moveLast:true})} />
        <ErrorWatcher visible={favoriteGifs.error !== ''}>{favoriteGifs.error}</ErrorWatcher>
      </FavoritesContainer>
    );
  }
}

const mapStateToProps = (state) => {
  const {favoriteGifs} = state;
  console.log('​mapStateToProps -> state', state);

  return {
    favoriteGifs
  };
};

export default connect(mapStateToProps, { getFavoriteGifsRequested })(Favorites);
