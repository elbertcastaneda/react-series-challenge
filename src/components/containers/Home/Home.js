import React, { Component } from 'react';
import { HomeContainer } from './Home.styled';
import { GiphyList } from '../../presentational';
import { connect } from 'react-redux';
import trendingGifsActions from '../../../actions/trendingGifs';
const { creators } = trendingGifsActions;
const { getTrendingGifsRequested } = creators;
class Home extends Component {
  componentDidMount () {
    const { getTrendingGifsRequested } = this.props;
    getTrendingGifsRequested();
  }

  render() {
    const {trendingGifs} = this.props;
    return (
      <HomeContainer>
        {/* {JSON.stringify(trendingGifs, null, 2)} */}
        <GiphyList dataSource={trendingGifs} />
      </HomeContainer>
    );
  }
}

const mapStateToProps = (state) => {
  console.log('â€‹mapStateToProps -> state', state);
  const {trendingGifs} = state;

  return {
    trendingGifs
  };
};

export default connect(mapStateToProps, { getTrendingGifsRequested })(Home);
