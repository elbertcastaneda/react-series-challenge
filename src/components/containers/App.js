import React, { Component } from 'react';
import {Route, Switch} from 'react-router-dom';
import {Menu, ItemMenu} from '../presentational';
import {AppContainer} from './App.styled';
import Home from './Home';

class App extends Component {
  render() {
    return (
      <AppContainer>
        <header>
          <Menu>
            <ItemMenu exact to='/'>Home</ItemMenu>
            <ItemMenu exact to='/favorites'>Favorites</ItemMenu>
            {/* <ItemMenu exact to='/gifs/123'>Detail</ItemMenu> */}
          </Menu>
        </header>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/favorites" component={() => <span>Favorites</span>} />
          <Route path="/gifs/:gif_id" component={
            ({match}) => <span>Detail of GIF {match.params.gif_id}</span>} />
          <Route component={() => <span>Not found</span>} />
        </Switch>
      </AppContainer>
    );
  }
}

export default App;
