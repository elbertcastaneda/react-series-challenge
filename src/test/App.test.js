import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Provider } from 'react-redux';
import {MemoryRouter as Router} from 'react-router-dom';
import store from '../store';
import App from '../components/containers/App';
import Home from '../components/containers/Home';
import Favorites from '../components/containers/Favorites';
import NotFound from '../components/containers/NotFound';

describe('App', () => {
  it('should render correctly', () => {
    const output = shallow(
      <App />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('Router and Redux provider', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );
    expect(wrapper.find(Home)).toHaveLength(1);
  });

  it('Move router to favorites', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router initialEntries={[ '/favorites' ]}>
          <App />
        </Router>
      </Provider>
    );
    expect(wrapper.find(Favorites)).toHaveLength(1);
  });

  it('Move router to non-existent route and its showed the not found page', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router initialEntries={[ '/anothe_none-existent_route' ]}>
          <App />
        </Router>
      </Provider>
    );
    expect(wrapper.find(NotFound)).toHaveLength(1);
    expect(wrapper.find(Favorites)).toHaveLength(0);
    expect(wrapper.find(Home)).toHaveLength(0);
  });
});
