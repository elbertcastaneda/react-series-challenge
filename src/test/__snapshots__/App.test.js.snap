// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<styled.div>
  <header>
    <Menu>
      <ItemMenu
        exact={true}
        to="/"
      >
        Home
      </ItemMenu>
      <ItemMenu
        to="/favorites"
      >
        Favorites
      </ItemMenu>
    </Menu>
  </header>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/favorites"
    />
    <Route
      component={[Function]}
      path="/gifs/:gif_id"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</styled.div>
`;
